### PR TITLE
doc: generate-changelog: add support for point releases

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Currently, the output of the [release-notes in the documentation](https://doc.riot-os.org/changelog.html) is broken. This should fix it.

### Testing procedure
Circle CI should pass and the changelog.html should look correct; point releases are listed in the Table of Contents.

### Issues/PRs references
None

### CODE REVIEW

**Comment by miri64** [comment]
[Here we go](https://4461-7370241-gh.circle-artifacts.com/0/doc/changelog.html). 

But mhhhh the weird diff pattern on top which caused me to make this PR and which provides a [weird entry in the deprecated list](https://4461-7370241-gh.circle-artifacts.com/0/doc/deprecated.html) still exists. I can't reproduce this locally with a newer version of Doxygen, so it seems to be a Doxygen bug. Nevertheless, parsing the point releases correctly still is an improvement.

**Comment by aabadie** [comment]
Please rebase to include the new required static-tests check.

File: doc/doxygen/generate-changelog.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def generate_changelog(template_filename, changelog_filename, output_filename):
          open(changelog_filename) as changelog, \
          open(output_filename, "w") as output:
         changelog_lines = []
-        release_title = re.compile(r"((RIOT-\d{4}\.\d{2} - Release Notes)|(Release 2013\.08))")
+        release_title = re.compile(r"((RIOT-\d{4}\.\d{2}(.\d+)? - Release Notes)|(Release 2013\.08))")
```
Comment (aabadie) [inline doc/doxygen/generate-changelog.py:19]: Should you also escape the `.` before `\d+` ?

```suggestion
        release_title = re.compile(r"((RIOT-\d{4}\.\d{2}(\.\d+)? - Release Notes)|(Release 2013\.08))")
```

(same comment for the other regexp change).

Comment (miri64) [reply doc/doxygen/generate-changelog.py:19]: True, will fix!

Comment (miri64) [reply doc/doxygen/generate-changelog.py:19]: Done

**Comment by aabadie** [comment]
The changelog indeed contains the point release now. Please squash!

**Comment by miri64** [comment]
Squashed

**Comment by aabadie** [comment]
Please rebase another time

**Review by aabadie — APPROVED** [review]
All good to me. The changelog is now available for point releases.

ACK

**Comment by miri64** [comment]
Rebased to include new checks.

**Pull request merged**